### PR TITLE
feat(48702): Emit for dynamic import (import()) when target >= ES2020 and module == None

### LINKED ISSUE #48702: Emit for dynamic import (`import()`) when target >= ES2020 and module == None
# Bug Report

We unconditionally transform dynamic import (`import()`) regardless of the current script target when the module kind is None. This is due to us using the `CommonJS` module emit as a fallback for `None`, and is causing an issue for VSCode while trying to implement support for TS Server plugins on the web via WebWorker.

I propose that we change the emit behavior for `import()` when the module target is `None` to preserve `import()`. Note that this would constitute a breaking change, but I imagine the number of affected projects (those currently using both `--module None` and `import()`) would be vanishingly small.

### 🔎 Search Terms

dynamic import module none node transform emit outFile

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?target=99&module=0#code/MYewdgzgLgBAHjAvDAlgWwA4gE5QBQBEAZiCAQJRA)

### 💻 Code

```ts
// @target: es2020
// @module: none
// @outFile: foo.js
const foo = import("foo");
```

### 🙁 Actual behavior

Transforms `import()` into a call to `require()`

### 🙂 Expected behavior

Does not transform `import()`.

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #48702

### CODE REVIEW

File: tests/baselines/reference/moduleNoneDynamicImport(target=es2015).js [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+//// [a.ts]
+const foo = import("./b");
+
+//// [b.js]
+export default 1;
+
+
+//// [a.js]
+const foo = Promise.resolve().then(() => require("b"));
```
Comment (DanielRosenwasser) [inline tests/baselines/reference/moduleNoneDynamicImport(target=es2015).js:11]: Isn't the request to make sure this preserves the `import`?

Comment (a-tarasyuk) [reply tests/baselines/reference/moduleNoneDynamicImport(target=es2015).js:11]: @DanielRosenwasser This file is related to target `es2015`. I thought the import should be preserved for `target >= es2020` and `module None`. Or am I wrong?

**Review by sandersn — APPROVED** [review]

**Comment by sandersn** [comment]
Looks good to me. Is this what you were thinking of @rbuckton ?

**Comment by a-tarasyuk** [comment]
@rbuckton Could you please review this PR?

**Comment by a-tarasyuk** [comment]
@rbuckton Could you review these changes?

**Review by rbuckton — APPROVED** [review]

**Pull request merged**